model/task/task.go

```diff
@@ -2048,25 +2061,38 @@ func topologicalSort(tasks []Task) ([]Task, error) {
 	return sortedTasks, nil
 }
 
-// DeactivateTask will set the ActivatedBy field to the caller and set the active state to be false and deschedule the task
-func (t *Task) DeactivateTask(caller string) error {
-	t.ActivatedBy = caller
-	t.Activated = false
-	t.ScheduledTime = utility.ZeroTime
-
-	return DeactivateTasks([]Task{*t}, true, caller)
-}
-
 func DeactivateTasks(tasks []Task, updateDependencies bool, caller string) error {
 	taskIDs := make([]string, 0, len(tasks))
 	for _, t := range tasks {
+		// Deactivating a deactivated task is a noop.
+		if !t.Activated {
+			continue
+		}
 		if t.DisplayOnly {
 			taskIDs = append(taskIDs, t.ExecutionTasks...)
 		}
 		taskIDs = append(taskIDs, t.Id)
 	}
 
-	_, err := UpdateAll(
+	depTasksToUpdate, depTaskIDsToUpdate, err := getDependencyTasksToUpdate(taskIDs, updateDependencies)
```

Comment:
Same rationale as this [comment](https://github.com/evergreen-ci/evergreen/pull/7765/files#r1579826112), but for deactivating tasks